### PR TITLE
Add hab-butterfly

### PR DESCRIPTION
**Pull request by adamhjk**
gif-keyboard-6291493242280433532

Adds a hab-butterfly CLI, which is called by `hab` if you issue a
command that needs to access the ring.

Adds the `Client` to butterfly, which allows you to send rumors directly
to a Butterfly server, without becoming a member first.

Currently this supports only the `hab config apply` command; a separate
PR will include the support for configuration files.

This leaves some warnings in place that will be cleaned up in the second
PR, as the code paths may well get resurrected.

Signed-off-by: Adam Jacob <adam@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/hab-butterfly/src/command/file.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    pub fn start(ui: &mut UI,
+                 peers: &Vec<String>,
+                 ring_key: Option<&SymKey>,
+                 user_pair: &BoxKeyPair,
+                 service_pair: &BoxKeyPair,
+                 number: u64,
+                 file_path: &Path)
+                 -> Result<()> {
+        //         try!(ui.begin(format!("Uploading file {}", &file_path.display())));
```
Comment (adamhjk) [inline components/hab-butterfly/src/command/file.rs]: This is an intentional place-holder

**Comment by reset** [comment]
@adamhjk Exactly as expected! Good work :smile:

gif-keyboard-2524626301418189099

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/178625372).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/178625372).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**